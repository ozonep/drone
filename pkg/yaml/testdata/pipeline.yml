kind: pipeline
name: default

platform:
  os: windows
  arch: arm64
  variant: 7
  version: 1803

workspace:
  base: /go
  path: src/github.com/drone/go-yaml

clone:
  disable: true
  depth: 50

steps:
- name: test
  image: golang
  pull: always
  shell: bash
  commands:
  - go get
  - go test
  environment:
    GOOS: windows
    GOARCH: arm
  privileged: true
  when:
    branch:
    - master
    - develop

- name: build
  build:
    image: octocat/hello-world
    context: .
    dockerfile: Dockerfile
    labels:
      foo: bar
      baz: qux
    cache_from:
    - octocat/knife-spoon

- name: push
  push: octocat/hello-world

services:
- name: redis
  image: redis:latest
  detach: true
  entrypoint: [ /bin/redis-server ]
  command: [ --port, 6380 ]
  ports:
  - 6380
  working_dir: /data
  volumes:
  - name: data
    path: /data
  failure: ignore
  environment:
    REDIS_USERNAME: foo
    REDIS_PASSWORD: bar

volumes:
- name: data
  temp: {}
- name: other
  host:
    path: /tmp/data

trigger:
  branch:
    include:
    - master
    - develop

depends_on:
- foo
- bar
