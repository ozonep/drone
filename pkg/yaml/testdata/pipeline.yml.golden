[
  {
    "kind": "pipeline",
    "name": "default",
    "clone": {
      "disable": true,
      "depth": 50
    },
    "depends_on": [
      "foo",
      "bar"
    ],
    "platform": {
      "os": "windows",
      "arch": "arm64",
      "variant": "7",
      "version": "1803"
    },
    "services": [
      {
        "command": [
          "--port",
          "6380"
        ],
        "detach": true,
        "entrypoint": [
          "/bin/redis-server"
        ],
        "environment": {
          "REDIS_PASSWORD": {
            "value": "bar"
          },
          "REDIS_USERNAME": {
            "value": "foo"
          }
        },
        "failure": "ignore",
        "image": "redis:latest",
        "name": "redis",
        "ports": [
          {
            "port": 6380
          }
        ],
        "volumes": [
          {
            "name": "data",
            "path": "/data"
          }
        ],
        "working_dir": "/data"
      }
    ],
    "steps": [
      {
        "commands": [
          "go get",
          "go test"
        ],
        "environment": {
          "GOARCH": {
            "value": "arm"
          },
          "GOOS": {
            "value": "windows"
          }
        },
        "image": "golang",
        "name": "test",
        "privileged": true,
        "pull": "always",
        "shell": "bash",
        "when": {
          "branch": {
            "include": [
              "master",
              "develop"
            ]
          }
        }
      },
      {
        "build": {
          "cache_from": [
            "octocat/knife-spoon"
          ],
          "context": ".",
          "dockerfile": "Dockerfile",
          "image": "octocat/hello-world",
          "labels": {
            "baz": "qux",
            "foo": "bar"
          }
        },
        "name": "build"
      },
      {
        "name": "push",
        "push": {
          "image": "octocat/hello-world"
        }
      }
    ],
    "trigger": {
      "branch": {
        "include": [
          "master",
          "develop"
        ]
      }
    },
    "volumes": [
      {
        "name": "data",
        "temp": {}
      },
      {
        "name": "other",
        "host": {
          "path": "/tmp/data"
        }
      }
    ],
    "workspace": {
      "base": "/go",
      "path": "src/github.com/drone/go-yaml"
    }
  }
]